src/mmrelay/meshtastic_utils.py

```
@@ -209,6 +216,42 @@ class BleExecutorDegradedError(Exception):
 subscribed_to_messages = False
 subscribed_to_connection_lost = False
 
+
+def ensure_meshtastic_callbacks_subscribed() -> None:
+    """Ensure Meshtastic pubsub callbacks are subscribed exactly once."""
+    global subscribed_to_messages, subscribed_to_connection_lost
```

Ruff (0.15.9)

[warning] Using the global statement to update `subscribed_to_messages` is discouraged

(PLW0603)